File: CSVExplorerJava/app/src/main/java/com/example/csvexplorerjava/service/AppDB.java
package com.example.csvexplorerjava.service;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import com.example.csvexplorerjava.dao.RowDao;
import com.example.csvexplorerjava.entity.RowEntity;


@Database(entities = {RowEntity.class}, version = 1, exportSchema = false)
public abstract class AppDB extends RoomDatabase {

    public abstract RowDao rowDao();

    private static volatile AppDB INSTANCE;

    public static AppDB get(Context context) {
        if (INSTANCE != null) return INSTANCE;
        synchronized (AppDB.class) {
            if (INSTANCE == null) {
                INSTANCE = Room.databaseBuilder(
                        context.getApplicationContext(),
                        AppDB.class,
                        "csv_exported.db"
                ).build();
            }
        }
        return INSTANCE;
    }
}

File: CSVExplorer/app/src/main/java/com/example/csvexplorer/service/AppDB.kt
package com.example.csvexplorer.service


import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import com.example.csvexplorer.dao.RowDao
import com.example.csvexplorer.entity.RowEntity

@Database(
    entities = [RowEntity::class],
    version = 1,
    exportSchema = false
)
abstract class AppDB : RoomDatabase() {

    abstract fun rowDao(): RowDao

    companion object {
        @Volatile private var INSTANCE: AppDB? = null

        fun get(context: Context): AppDB {
            return INSTANCE ?: synchronized(this) {
                INSTANCE ?: Room.databaseBuilder(
                    context.applicationContext,
                    AppDB::class.java,
                    "csv_exported.db"
                ).build().also { INSTANCE = it }
            }
        }
    }
}
